### PR TITLE
Modernizes/Fixes main stats project

### PR DESCRIPTION
**Pull request by astralbodies**
There were several issues with the main stats project. This project is only used for unit test execution and anyone out there in the world that wants to manually install our stats framework into their app (so, nobody). Originally the project had a static library target and it was updated to a dynamic library target a while back to support frameworks. We never updated the Podfile though to use frameworks. Surprisingly this hadn't broken anything until we added Swift to the project.

1. CocoaLumberjack gives you the ability to override the macro definition `LOG_LEVEL_DEF` in order to let you use your own static Int to define a logging level for the project. When enabling modules, however, these macros are scoped to the module (sort of) and overrides don't work right. Scrapped the custom definition and fell back to using `ddLogLevel`.
2. I deleted and re-added the framework and unit testing targets. This modernizes all of the build settings to be appropriate to a dynamic library rather than a static library. I'm a moron when it comes to compiler and linker configuration so this was the least stabby option.
3. Updated a few classes to use the more modern `@import ModuleName;` syntax in Objective-C because of some compiler warnings.

We'll have to most likely update the shared repo as well to fix the CocoaLumberjack issue.

Needs Review: @bummytime, @jleandroperez, @koke

### CODE REVIEW

File: WordPressCom-Stats-iOS/WordPressCom-Stats-iOS-Prefix.pch [hunk 1]
```diff
@@ -7,8 +7,7 @@
 #ifdef __OBJC__
 
 #import <Foundation/Foundation.h>
-#define LOG_LEVEL_DEF WordPressComStatsLogLevel
 #import <CocoaLumberjack/CocoaLumberjack.h>
-static const DDLogLevel WordPressComStatsLogLevel = DDLogLevelWarning;
+static const DDLogLevel ddLogLevel = DDLogLevelWarning;
```
Comment (koke) [inline WordPressCom-Stats-iOS/WordPressCom-Stats-iOS-Prefix.pch]: As discussed on Slack, this won't get properly override by the app, so I did #392

**Comment by bummytime** [comment]
> I deleted and re-added the framework and unit testing targets. This modernizes all of the build settings to be appropriate to a dynamic library rather than a static library. I'm a moron when it comes to compiler and linker configuration so this was the least stabby option.

My kind of fix ;-) I also :+1: @koke's addition to this PR. 

For grins, I tried integrating this branch into WPiOS. Looks like the new `Common` folder is not coming along for the ride:

screen shot 2016-03-16 at 4 38 34 pm

Not sure if we actually need to create a new subspec in the stats podspec for `Common` since we ref this pod via its component specs in WPiOS or something else...

**Comment by koke** [comment]
So much fun... I've tried a different thing in a0ec243. Now there's nothing to cause problems in the PCH other than Foundation. I've tested the test project, StatsDemo and the app with this and it seems to build and work as expected.

**Comment by astralbodies** [comment]
I like it @koke - lets roll with it.

Thanks everyone for the review!

**Pull request merged**